code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/CarouselPushTemplate.kt

```diff
@@ -108,23 +95,16 @@ internal open class CarouselPushTemplate : AEPPushTemplate {
                 val jsonArray = JSONArray(carouselItemsString)
                 for (i in 0 until jsonArray.length()) {
                     val item = jsonArray.getJSONObject(i)
-                    val imageUri = item.getString(PushTemplateConstants.CarouselItemKeys.IMAGE)
-                    val captionText =
-                        item.optString(PushTemplateConstants.CarouselItemKeys.TEXT, "")
-                    val interactionUri =
-                        item.optString(PushTemplateConstants.CarouselItemKeys.URI, "")
+                    val imageUri = item.getString(CarouselItemKeys.IMAGE)
+                    val captionText = item.optString(CarouselItemKeys.TEXT, "")
+                    val interactionUri = item.optString(CarouselItemKeys.URI, "")
                     carouselItems.add(
-                        CarouselItem(
-                            imageUri,
-                            captionText,
-                            interactionUri
-                        )
+                        CarouselItem(imageUri, captionText, interactionUri)
                     )
                 }
             } catch (e: JSONException) {
                 Log.debug(
```

Comment:
This should throw exception in case Json is invalid @rymorale can you verify this please, as we are now merging the logic to one constructor. e.g. we are doing it in ProductCatalogPushTemplate.kt#92

Reply:
Isn't the current code still throwing the exception? and gives out empty carouselItems list back from this method ?

Sorry I might have mis understood the ask here?